### PR TITLE
[BACKPORT-2.1][SPARK-19318][SPARK-22041][SQL] Docker test case failure: `SPARK-16625: General data types to be mapped to Oracle`

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

This PR is backport of https://github.com/apache/spark/pull/16891 to Spark 2.1.

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by gatorsmile** [comment]
Have you run the docker test?

**Comment by wangyum** [comment]
@gatorsmile Yes, Docker integration tests passed.

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 1]
```diff
@@ -82,7 +107,7 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
     // A value with fractions from DECIMAL(3, 2) is correct:
     assert(row.getDecimal(1).compareTo(BigDecimal.valueOf(1.23)) == 0)
     // A value > Int.MaxValue from DECIMAL(10) is correct:
-    assert(row.getDecimal(2).compareTo(BigDecimal.valueOf(9999999999l)) == 0)
+    assert(row.getDecimal(2).compareTo(BigDecimal.valueOf(9999999999L)) == 0)
```
Comment (gatorsmile) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: revert this back

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 2]
```diff
@@ -63,15 +63,40 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
...
+    conn.prepareStatement(
+      "INSERT INTO numerics VALUES (4, 1.23, 9999999999)").executeUpdate()
+    conn.commit()
   }
 
-
-  test("SPARK-16625 : Importing Oracle numeric types") { 
-    val df = sqlContext.read.jdbc(jdbcUrl, "numerics", new Properties);
+  test("SPARK-16625 : Importing Oracle numeric types") {
+    val df = sqlContext.read.jdbc(jdbcUrl, "numerics", new Properties)
```
Comment (gatorsmile) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: revert this back

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala [hunk 3]
```diff
@@ -913,6 +913,25 @@ class JDBCSuite extends SparkFunSuite
       "dbtable" -> "t1",
       "numPartitions" -> "10")
     assert(new JDBCOptions(parameters).asConnectionProperties.isEmpty)
-    assert(new JDBCOptions(new CaseInsensitiveMap(parameters)).asConnectionProperties.isEmpty)
+    assert(new JDBCOptions(CaseInsensitiveMap(parameters)).asConnectionProperties.isEmpty)
+  }
+
+  test("SPARK-19318: jdbc data source options should be treated case-insensitive.") {
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala]: Where is the test case `test("SPARK-19318: Connection properties keys should be case-sensitive.")`?

**Comment by gatorsmile** [comment]
LGTM pending test.

**Comment by wangyum** [comment]
retest this please.

**Comment by gatorsmile** [comment]
Thanks! Merged to 2.1. 

Could you close this PR?

**Pull request closed without merging**